tests/test_servicecatalog.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2020 Capital One Services, LLC
+# Copyright The Cloud Custodian Authors.
+# SPDX-License-Identifier: Apache-2.0
+
+from .common import BaseTest
+import time
+from c7n.exceptions import PolicyValidationError
+
+
+class TestServiceCatalog(BaseTest):
+
+    def test_portfolio_delete(self):
+        session_factory = self.replay_flight_data("test_portfolio_delete")
+        p = self.load_policy(
+            {
+                "name": "servicecatalog-portfolio-delete",
+                "resource": "catalog-portfolio",
+                "filters": [{"tag:name": "pratyush"}],
+                "actions": [{"type": "delete"}],
+            },
+            session_factory=session_factory)
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
+        self.assertEqual(resources[0]['Id'], 'port-tb63f2b33cska')
+        if self.recording:
+            time.sleep(10)
+        client = session_factory().client("servicecatalog")
+        portfolios = client.list_portfolios()
+        self.assertFalse('port-tb63f2b33cska' in [p.get(
+            'Id') for p in portfolios.get('PortfolioDetails')])
+
+    def test_portfolio_cross_account_remove(self):
```

Comment:
You are missing a test case for the specific account numbers with `remove-shared-accounts`.

It would be good to have at least two resources that match, where only one has the specified account number.

Reply:
was also wondering if we want to do this as set-shared-accounts with remove/add options

Reply:
`set-shared-accounts` with remove/add option sounds good to me. I'll go ahead and make the change and will also add a test case around that behavior

Reply:
One interesting thing I found out wrt the `add` option is that the share request would have to be accepted at the other end. @kapilt Do you still think it might be worth adding it here, keeping in mind it won't be a full proof feature?

Reply:
ah, let's skip it, we don't want to two way hand shakes across arbitrary accounts in policies.

Reply:
cool, added a test case as per Tim's suggestion